### ISSUE TITLE
Headphone amp driver doesn't work for inputs without a decimal point

### ISSUE THREAD

**Issue by tvannoy** [fpga-open-speech-tools/component_library#5]
When writing to the volume register, values without a decimal point do not get parsed correctly. For example, writing -20 to the volume register results in 
```
[  914.313055] Maximum attenuation exceeded.
[  914.313062] Setting attenuation to -100 dB.
```
being printed to the kernel log.

In light of this, perhaps all of our drivers should use shared (and tested) string parsing/conversion functions.

**Comment by fe-wickham** [comment]
This would be stemming from a bug in the set_fixed_num function, correct? As you are seeing, it looks like the function just doesn't handle there not being a decimal point.

**Comment by tvannoy** [comment]
Yes, that should be the case. `set_fixed_num` in custom_functions.h should have a working implementation. I'd vote for including that header in this driver to avoid duplicating code.
